src/autocomplete.ts

```diff
@@ -40,6 +42,25 @@ export default class Autocomplete {
       this.feedback.setAttribute('aria-atomic', 'true')
     }
 
+    // if clearButton is not a button, show error
+    if (this.clearButton && this.clearButton.tagName.toLowerCase() !== 'button') {
+      const buttonErrorMessage = `Accessibility violation: ${this.clearButton.tagName} provided for clear button. Please use a "button" element.`
+      const buttonErrorText = '\u26a0 Error: See console'
+      const buttonError = document.createElement('span')
+      buttonError.setAttribute('style', 'color:#92140C')
+      buttonError.textContent = buttonErrorText
+      this.clearButton.parentNode?.replaceChild(buttonError, this.clearButton)
+      this.clearButton = buttonError
+      throw new Error(buttonErrorMessage)
+    }
+
```

Comment:
We do not want to be presenting errors to (potentially) the user. I don't think this should be here.

Instead of reporting accessibility violations within the main code here, we might want to add an additional file which can provide custom rules to [Axe](https://github.com/dequelabs/axe-core). Something like `axe.ts` with the following:


```ts
export default [
  {
    id: 'auto-complete-element-clear-button-is-button',
    impact: 'critical',
    selector: 'auto-complete-element [id$="clear"]:not(button)',
    excludeHidden: false,
    metadata: {
      description: "Auto-Complete clear elements must be a "button" element",
      helpUrl: "https://github.com/github/auto-complete-element"
    }
  }
]
```

We can then load this into the existing devtools we have for GitHub (which includes axe-core) and it can report on these types of violations during development without impacting the user.

Reply:
Let's set up a sync to chat about options for this. I don't want end-users to see errors either, but this package doesn't include Axe, so I'd like to avoid relying on it. It *is* open source and could be used anywhere. I'll DM you.